### DISCUSSION TITLE
Prisma Client was generated for "debian-openssl-1.1.x", but the actual deployment required "debian-openssl-3.0.x".

### DISCUSSION THREAD

**Discussion by mrhappyma** [prisma/prisma#21508] [Q&A]
### Bug description

Using docker to build and run my project. Building goes fine but I'm getting this error on starting the container
```
Invalid `prisma.game.findFirstOrThrow()` invocation:


Prisma Client could not locate the Query Engine for runtime "debian-openssl-3.0.x".

This happened because Prisma Client was generated for "debian-openssl-1.1.x", but the actual deployment required "debian-openssl-3.0.x".
Add "debian-openssl-3.0.x" to `binaryTargets` in the "schema.prisma" file and run `prisma generate` after saving it:

prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-3.0.x"]
}

The following locations have been searched:
  /node_modules/.prisma/client
  /node_modules/@prisma/client
  /home/dominic/Documents/ups-and-downs/node_modules/@prisma/client
  /tmp/prisma-engines
  /prisma
      at new e (/node_modules/@prisma/client/runtime/library.js:26:1937)
      at handleRequestError (/node_modules/@prisma/client/runtime/library.js:123:7089)
      at handleAndLogRequestError (/node_modules/@prisma/client/runtime/library.js:123:6200)
      at /node_modules/@prisma/client/runtime/library.js:123:5920
      at processTicksAndRejections (:55:76)

```
here's my dockerfile:
```
FROM ubuntu:jammy
COPY . .
RUN apt-get update -y; apt-get install curl unzip openssl build-essential -y
RUN curl -fsSL https://bun.sh/install | bash
RUN ~/.bun/bin/bun i
CMD ["/root/.bun/bin/bun", "run", "src/index.ts"]
EXPOSE 3000
```

this happens with and without the extra `binaryTargets` field added in my schema

### Environment & setup

- OS: Ubuntu in Docker
- Database: Postgres

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.4.2
@prisma/client          : 5.4.2
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/schema-engine-debian-openssl-1.1.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Studio                  : 0.494.0
```
`prisma -v` ran in project but not within container

**Comment by ludralph** [comment]
Hi @mrhappyma 👋 

Thank you for raising this question.

Can you please specify the `binaryTargets` as shown below? Take a look at this section of the [documentation](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#specify-custom-binarytargets-to-ensure-compatibility-with-the-os)
```
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["debian-openssl-3.0.x"]
}
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Ralle** [reply]
I just hit this same snag on the `node:20-slim` image. My default the binaryTargets is `["native"]` which auto-selects based on the OS. How come it doesn't auto-select `debian-openssl-3.0.x`?

**Reply by mrhappyma** [reply]
This is where I am confused. I add the target as instructed by the error, and yet still get the same error that it was generated for the wrong target.
```
Prisma Client could not locate the Query Engine for runtime "debian-openssl-3.0.x".

This happened because Prisma Client was generated for "debian-openssl-1.1.x", but the actual deployment required "debian-openssl-3.0.x".
Add "debian-openssl-3.0.x" to `binaryTargets` in the "schema.prisma" file and run `prisma generate` after saving it:

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-3.0.x"]
}
```

**Reply by thomas779** [reply]
@mrhappyma @Ralle encountering this same issue as well.

did this issue show up?
```
PrismaClientInitializationError: Unable to require(`/app/prisma/generated/client/libquery_engine-debian-openssl-3.0.x.so.node`).
Prisma cannot find the required `libssl` system library in your system. Please install openssl-3.0.x and try again.
Details: libssl.so.3: cannot open shared object file: No such file or directory
```

**Reply by mattgraphlan** [reply]
for the internet searchers, I had a related bug where my github action was not downloading the additional engine binaries unless I set the env var `PRISMA_CLI_BINARY_TARGETS` 
 https://github.com/prisma/prisma/issues/23009

**Comment by punkbit** [comment]
For future readers finding this issue on GitHub Actions, if you'd like to stick with `openssl-1.1.x` until you have more time to troubleshoot, you can remove `ubuntu-latest` and use a fixed version, e.g., `ubuntu-20.04`.

```
runs-on: ubuntu-20.04
```

By doing this, the runner environment is of a fixed version and you won't be surprised by latest runner versions, which might cause this or other issues.

Ref:
- GitHub's runner `22.04` has `OpenSSL 3.0.2-0ubuntu1.17`
https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
